src/app/hooks.ts

```diff
@@ -0,0 +1,5 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from './store';
+
+export const useAppSelector = useSelector.withTypes<RootState>();
```

Comment:
The use of `useSelector.withTypes<RootState>()` is incorrect. The `useSelector` hook does not have a `withTypes` method. Instead, you can directly use `useSelector` with a type assertion or by using a typed selector function.